src/main/scala/org/graphframes/lib/ConnectedComponents.scala

```diff
@@ -336,8 +338,11 @@ object ConnectedComponents extends Logging {
 
         // remove previous checkpoint
         if (iteration > checkpointInterval) {
-          FileSystem.get(sc.hadoopConfiguration)
-            .delete(new Path(s"${checkpointDir.get}/${iteration - checkpointInterval}"), true)
+          val path = new Path(s"${checkpointDir.get}/${iteration - checkpointInterval}")
+          if (fileSystem == null) {
```

Comment:
although to fix #162, we will likely need to store the FileSystem in order to clean up the very last checkpoint saved.

but more can come then